sdk/client/client.go

```diff
@@ -327,14 +327,11 @@ func (c *Client) Execute(ctx context.Context, req *Request) (*Response, error) {
 		return nil, fmt.Errorf("req.Request was nil")
 	}
 
-	// at this point we're ready to send the HTTP Request, as such let's get the Authorization token
-	// and add that to the request
+	// Set Authorization and X-Ms-Authorization-Auxiliary headers
```

Comment:
Doing this will cause issues with Storage Data Plane which is _super_ picky about the headers it supports

Perhaps Client should have a `ConfigureAuthorizerForRequest` function/method defined against the struct which the implementations can pass-in?

Reply:
Makes sense, I've added this and set a custom authorization function for `ResourceManagerClient`. Where such custom function is not specified, the base client will fall back to setting a standard `Authorization` header with a bearer token (i.e. its current behavior).